core/src/main/java/gov/nist/secauto/metaschema/core/metapath/function/library/FnIndexOf.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * SPDX-FileCopyrightText: none
+ * SPDX-License-Identifier: CC0-1.0
+ */
+
+package gov.nist.secauto.metaschema.core.metapath.function.library;
+
+import gov.nist.secauto.metaschema.core.metapath.DynamicContext;
+import gov.nist.secauto.metaschema.core.metapath.ISequence;
+import gov.nist.secauto.metaschema.core.metapath.MetapathConstants;
+import gov.nist.secauto.metaschema.core.metapath.function.FunctionUtils;
+import gov.nist.secauto.metaschema.core.metapath.function.IArgument;
+import gov.nist.secauto.metaschema.core.metapath.function.IFunction;
+import gov.nist.secauto.metaschema.core.metapath.item.IItem;
+import gov.nist.secauto.metaschema.core.metapath.item.atomic.IAnyAtomicItem;
+import gov.nist.secauto.metaschema.core.metapath.item.atomic.IBooleanItem;
+import gov.nist.secauto.metaschema.core.metapath.item.atomic.IIntegerItem;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * /**
+ * Implements <a href=
+ * "https://www.w3.org/TR/xpath-functions-31/#func-index-of">fn:index-of</a>
+ * functions. This implementation does not implement the three-arg variant with
+ * collation at this time.
+ */
+public final class FnIndexOf {
+  @NonNull
+  private static final String NAME = "index-of";
+  @NonNull
+  static final IFunction SIGNATURE_TWO_ARG = IFunction.builder()
+      .name(NAME)
+      .namespace(MetapathConstants.NS_METAPATH_FUNCTIONS)
+      .deterministic()
+      .contextDependent()
+      .focusIndependent()
+      .argument(IArgument.builder()
+          .name("seq").type(IAnyAtomicItem.class)
+          .zeroOrMore()
+          .build())
+      .argument(IArgument.builder()
+          .name("search")
+          .type(IAnyAtomicItem.class)
+          .one()
+          .build())
+      .returnType(IIntegerItem.class)
+      .returnOne()
+      .functionHandler(FnIndexOf::executeTwoArg)
+      .build();
+
+  @SuppressWarnings("unused")
+  @NonNull
+  private static ISequence<IIntegerItem> executeTwoArg(@NonNull IFunction function,
+      @NonNull List<ISequence<?>> arguments,
+      @NonNull DynamicContext dynamicContext,
+      IItem focus) {
+    ISequence<?> seq = arguments.get(0).asSequence();
```

Comment:
Blocker: Review request: most other functions peel out the first item in the sequence. I know this is poor form, I am not sure the best way to take the sequence "as-is" and defensively null check with a util function having read some util function docs. I expect to be dinged on this.

Reply:
You can use `arguments.get(0)` to get the actual sequence.